src/custom_flags/rustfix.rs

```diff
@@ -171,18 +117,124 @@ impl Flag for RustfixMode {
             });
         }
 
-        if !run {
-            return Ok(Some(cmd));
+        compile_fixed(config, build_manager, &fixed_paths)?;
+
+        if fixed.is_empty() {
+            Ok(Some(cmd))
+        } else {
+            Ok(None)
         }
+    }
+}
+
+fn fix(stderr: &[u8], path: &Path, filter: rustfix::Filter) -> anyhow::Result<Vec<String>> {
```

Comment:
For the future, when doing large refactorings like this, please split out the code movements into separate commits

Reply:
Sorry yeah, should've gone back to do that. Should I split it out still or leave it as is?

Reply:
Nah it's ok, but the Rebase will be harder

Reply:
Yeah there were indeed a lot of conflicts so I ended up reapplying it manually, stuck it in its own commit while I was there